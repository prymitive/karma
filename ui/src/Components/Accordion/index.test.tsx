import { mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import { Accordion, AccordionItem } from ".";

describe("<Accordion />", () => {
  it("matches snapshot", () => {
    const tree = mount(
      <Accordion>
        <AccordionItem text="title 1" content="item 1" />
        <AccordionItem text="title 2" content="item 2" defaultIsOpen />
        <AccordionItem text="title 1" content="item 1" />
      </Accordion>
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });
});

describe("<AccordionItem />", () => {
  it("doesn't render content by default", () => {
    const tree = mount(<AccordionItem text="title" content="content" />);
    expect(tree.text()).toBe("title");
  });

  it("doesn't render content when defaultIsOpen=false", () => {
    const tree = mount(
      <AccordionItem text="title" content="content" defaultIsOpen={false} />
    );
    expect(tree.text()).toBe("title");
  });

  it("renders content when defaultIsOpen=true", () => {
    const tree = mount(
      <AccordionItem text="title" content="content" defaultIsOpen={true} />
    );
    expect(tree.text()).toBe("titlecontent");
  });

  it("toggles content after header click", () => {
    const tree = mount(<AccordionItem text="title" content="content" />);
    expect(tree.text()).toBe("title");

    tree.find("button.accordion-button").simulate("click");
    expect(tree.text()).toBe("titlecontent");

    tree.find("button.accordion-button").simulate("click");
    expect(tree.text()).toBe("title");
  });
});
