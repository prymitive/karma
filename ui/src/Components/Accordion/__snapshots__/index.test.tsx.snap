// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> matches snapshot 1`] = `
"
<div class=\\"accordion\\">
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button collapsed\\"
              type=\\"button\\"
      >
        title 1
      </button>
    </h2>
    <div class=\\"accordion-collapse \\">
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        title 2
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        item 2
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button collapsed\\"
              type=\\"button\\"
      >
        title 1
      </button>
    </h2>
    <div class=\\"accordion-collapse \\">
    </div>
  </div>
</div>
"
`;
